### PR TITLE
Add canner storage

### PR DESCRIPTION
**Pull request by chihsuan**
Issue: https://github.com/Canner/canner/issues/97

### CODE REVIEW

File: packages/canner-storage/src/ImgurStorage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  mashapeKey: string;
+  
+  constructor({clientId = '', mashapeKey = ''}: {clientId: string, mashapeKey?: string}) {
+    this.clientId = clientId;
+    this.mashapeKey = mashapeKey;
+  }
+
+  getUploadUrl() {
+    return {
+      url: IMGUR_IMAGE_API
```
Comment (abz53378) [inline packages/canner-storage/src/ImgurStorage.js]: add mashape API
https://github.com/Canner/image-service-config/blob/master/src/ImgurService.js#L25
```js
if (this.maspapeKey)  {
  return MASHAPE_API;
} else {
  return IMGUR_IMAGE_API;
}
```

**Review by abz53378 — CHANGES_REQUESTED** [review]

File: packages/canner-storage/src/ImgurStorage.js (hunk 1 above)
Comment (chihsuan) [reply packages/canner-storage/src/ImgurStorage.js]: @abz53378: Updated, thanks!

File: packages/canner-storage/src/FirebaseClientStorage.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      throw new Error("firebase client instance is required.");
+    }
+    this.firebase = firebase;
+    this.dir = '';
+  }
+
+  upload(file: File, options: UploadOptions, onProgress: OnProgressType) {
+    const images = this.firebase
+      .storage()
+      .ref(this.dir)
```
Comment (abz53378) [inline packages/canner-storage/src/FirebaseClientStorage.js]: Please remove `this.dir`, just call `.ref()` here.

File: packages/canner-storage/src/ImgurStorage.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  getUploadUrl() {
+    return {
+      url: this.mashapeKey ? MASHAPE_API : IMGUR_IMAGE_API
+    };
+  }
+
+  upload(file: File, options: UploadOptions, onProgress: OnProgressType) {
+    let UPLOAD_API = IMGUR_IMAGE_API;
```
Comment (abz53378) [inline packages/canner-storage/src/ImgurStorage.js]: I think we can get the `UPDATE_API` with calling `this.getUploadUrl()` here.

**Review by abz53378 — CHANGES_REQUESTED** [review]

File: packages/canner-storage/src/FirebaseClientStorage.js (hunk 2 above)
Comment (chihsuan) [reply packages/canner-storage/src/FirebaseClientStorage.js]: Removed. Thanks!

File: packages/canner-storage/src/ImgurStorage.js (hunk 3 above)
Comment (chihsuan) [reply packages/canner-storage/src/ImgurStorage.js]: Good point. I've changed it. Thanks!

**Review by abz53378 — APPROVED** [review]

**Pull request merged**